kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableImpl.java

```diff
@@ -47,14 +47,16 @@ public static Table forPath(TableClient tableClient, String path)
 
     private final Path logPath;
     private final Path dataPath;
+    private final SnapshotManager snapshotManager;
 
     public TableImpl(Path logPath, Path dataPath) {
         this.logPath = logPath;
         this.dataPath = dataPath;
+        this.snapshotManager = new SnapshotManager();
```

Comment:
@vkorukanti we should have a design philosophy for when and how we construct objects.
what if this constructor eventually gets expensive? etc.

Reply:
https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java#L179

Reply:
Sorry, pardon?